modules/core/src/parallel.cpp

```diff
@@ -779,8 +773,51 @@ static inline int getNumberOfCPUsImpl()
 }
 #endif
 
+#ifdef __ANDROID__
+static inline int getNumberOfCPUsImpl()
+{
+   FILE* cpuPossible = fopen("/sys/devices/system/cpu/possible", "r");
+   if(!cpuPossible)
+       return 1;
+
+   char buf[2000]; //big enough for 1000 CPUs in worst possible configuration
+   char* pbuf = fgets(buf, sizeof(buf), cpuPossible);
+   fclose(cpuPossible);
+   if(!pbuf)
+      return 1;
+
+   return getNumberOfCPUsImplFromFileStr(pbuf);
+}
+#endif
+
+#if (defined __linux__ || defined __GLIBC__ || defined __HAIKU__ || defined __EMSCRIPTEN__) && !defined __ANDROID__
+static inline unsigned getNumberOfCPUsImplLinux()
+{
+   FILE* cpuPossible = fopen("/sys/fs/cgroup/cpuset/cpuset.cpus", "r");
```

Comment:
This is cgroup v1 files, for v2 it will be different:
http://man7.org/linux/man-pages/man7/cgroups.7.html
https://wiki.archlinux.org/index.php/cgroups

Shouldn't we first determine cgroup of current process and then use it in the path?

Reply:
Does HAIKU and EMSCRIPTEN have cgroups at all?

Reply:
Thanks for the review @mshabunin, As per [this](https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html#mounting), they are backward compatible and we get the info we need from v1 itself. Open to change if required as I do not know much about it. 


> Does HAIKU and EMSCRIPTEN have cgroups at all?

I am not sure about this, but if the file does not exist, we handle that case by returning 0. Again open to removing it if needed, did not find much data on this.

Reply:
I think it is better to restore previous behavior for HAIKU and EMSCRIPTEN cases.

Reply:
I've rebooted with `cgroup_no_v1=all` option and this path is not available anymore:
```
$ ls /sys/fs/cgroup/
blkio        cpuset   freezer  memory            perf_event  rdma     unified
cpu,cpuacct  devices  hugetlb  net_cls,net_prio  pids        systemd
```
I think we should have fallback to previous method if it is not readable.

Reply:
Oh cool, sounds good. I'll remove HAIKU and EMSCRIPTEN. I'll return 0 if the file is absent and handle.

Reply:
I haven't removed Haiku and emscripten to avoid extra # ifs and a possibility of the file existing, but handled it logically.